api/domain_handlers.go

```diff
@@ -381,3 +382,70 @@ func (api API) Remove(w http.ResponseWriter, r *http.Request) {
 
 	writeJSONOrBust(w, issues)
 }
+
+// RemoveIneligibleDomains runs eligibility checks on domains present in the
+// database and change the status to PendingAutomatedRemoval if the domain
+// does not follow the requirements for more than 2 crawls
+
+// Example: GET /removeineligibledomains?
+func (api API) RemoveIneligibleDomains(w http.ResponseWriter, r *http.Request) {
+
+	var ineligibleDomains []database.IneligibleDomainState
+	var deleteEligibleDomains []string
+
+	domains, err := api.database.AllDomainStates()
+	if err != nil {
+		msg := fmt.Sprintf("Internal error: could not retrieve domains. (%s)\n", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+		return
+	}
+	// Store ineligible domains in slice
+	for _, d := range domains {
+		if d.Policy == preloadlist.Bulk18Weeks || d.Policy == preloadlist.Bulk1Year {
+
+			_, issues := api.hstspreload.EligibleDomain(d.Name, d.Policy)
```

Comment:
(I missed this in the previous review comments.) This function takes some time to scan domains and return. Currently, these are done sequentially, which given the number of domains could result in this endpoint timing out. For this PR, how about adding a TODO to consider parallelizing this operation?